app/controllers/timeline_controller.rb

```diff
@@ -32,12 +32,14 @@ def set_lists
 
   def paginate_timeline
     @timeline = yield(last_id: params[:last_id])
-    @timeline_last_id = @timeline.map(&:id).min
+    timeline_ids = @timeline.map(&:id)
+    @timeline_last_id = timeline_ids.min
     @more_data_available = !yield(last_id: @timeline_last_id, size: 1).count.zero?
+    subscribed_answer_ids = Subscription.where(user: current_user, answer_id: timeline_ids).pluck(:answer_id) if user_signed_in?
```

Comment:
`if user_signed_in?` is redundant here, as all of these methods are `:authenticate_user!`